sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/subquery.scala

```diff
@@ -274,7 +278,9 @@ object SubExprUtils extends PredicateHelper {
     plan match {
       case Filter(cond, child) =>
         val correlated = AttributeSet(splitConjunctivePredicates(cond)
-          .filter(containsOuter) // TODO: can remove this line to allow e.g. where x = 1 group by x
+          .filter(
+            SQLConf.get.getConf(SQLConf.SCALAR_SUBQUERY_ALLOW_GROUP_BY_COLUMN_EQUAL_TO_CONSTANT)
```

Comment:
I'm not sure we need this config and this restriction is newly added. It's OK to relax this un-released restriction.

Reply:
Yea I think we can remove this config.

Reply:
Actually the restriction isn't new - it was there before for many years. So I added the config as per the usual flag any surface area change. Can remove it if we don't need it though.